### DISCUSSION TITLE
Idle hobbies on a Sunday afternoon

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#325] [Show and tell]
![336AB1D9-6330-47BD-97BD-4996EFA36323](https://user-images.githubusercontent.com/12018881/206920913-67e29d31-30d1-46cc-9445-16ab3ff768fb.png)
This with ISH, a 32-bit Alpine on an 586 emulator, on a non-jailbroken iPad!

**Comment by mgrossmann** [comment]
Hey René,

Looks like a lot of fun. 👌
I had to do some tax stuff, this weekend. 😩

Best, Mike

René Vincent Jansen ***@***.***> schrieb am So. 11. Dez. 2022
um 19:10:

> [image: 336AB1D9-6330-47BD-97BD-4996EFA36323]
> <https://user-images.githubusercontent.com/12018881/206920913-67e29d31-30d1-46cc-9445-16ab3ff768fb.png>
> This with ISH, a 32-bit Alpine on an 586 emulator, on a non-jailbroken
> iPad!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/adesutherland/CREXX/discussions/325>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAF7Q7L7FDAEW5EYSCR5V7DWMYKJRANCNFSM6AAAAAAS3D7T6A>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Von Gmail Mobile gesendet

**Reply by rvjansen** [reply]
Tax is horrible, and I say that as someone who has worked at the tax offices in two separate countries. I'll gladly subscribe to any religion who has rules against doing taxes on a Sunday ... o wait ... . In Amsterdam before the war there were Jewish families who employed a 'goy' (non-Jew) to operate the light switches on a Sabbath, when it was not allowed for orthodox people to do so. I guess this is why I employ a bookkeeper to do taxes.

Anyway, getting this to run on the iPad was an object lesson in how accidents hide in small corners. The whole system is C99 and only requires a compiler, cmake, rexx and bison so it must be a walk in the park to have that run on a mainstream 32-bit Linux, so one might think.

Unfortunately the cmake req of 3.18 in the lib/rxfns directory is true; 3.14.5 fails and 3.20 succeeds. Also, Ninja needs to be quite new or it complains about circular dependencies. I did the rest with make and by just typing in stuff. When I took the screenshot I was in the middle of this but I thought you were all going to sleep before I finish this.

So I did a few compiles and links by hand and took the screenshot. Now to have it all smooth, and having burned through the battery already, I decided to run a Docker image, install the dependencies and do a cross-compile, so I could have only the executables and the BIF library on the iPad. (You understand I am doing this for when civilisation ends and there is no server I can connect to.)

To my not-so-small surprise, only the rxas executable worked. All others complained a lot about missing __time64 and other 64-bit (resolution) time calls. Turns out that the maintainers of Alpine made a boo-boo just between the release on the iPad and the Docker image. I then downloaded the ISH app to the Apple M1 - things like that work nowadays. But alas, git does choke on processing delta's; for an unknown reason. On the Docker image the same thing happens (it might be broken, btw) but fortunately over there I can bind-mount a working checkout.

Follows an hour or so where I learn more about the APK (Alpine Package Manager) than I honestly want to know. In that time, I have downloaded a new ninja, a new cmake and rebuilt - drat, the same errors. Turns out that I also needed to uplevel the g++ compiler to make everything work, and now everything indeed does, and is built on the iPad, in ISH, itself.

I think the lesson here is that we also must build on as many platforms as we can, see what happens and maybe ask Per-Olof from RexxLA to give us a hand to have all this automated on the Jenkins that runs all these VM's.

**Comment by MattCrBoom** [comment]
!esa
